tools/installer/lib/ide-setup.js

```diff
@@ -93,6 +97,643 @@ class IdeSetup extends BaseIdeSetup {
     }
   }
 
+  async setupOpenCode(installDir, selectedAgent, spinner = null, preConfiguredSettings = null) {
+    // Minimal JSON-only integration per plan:
+    // - If opencode.json or opencode.jsonc exists: only ensure instructions include .bmad-core/core-config.yaml
+    // - If none exists: create minimal opencode.jsonc with $schema and instructions array including that file
+
+    const jsonPath = path.join(installDir, 'opencode.json');
+    const jsoncPath = path.join(installDir, 'opencode.jsonc');
+    const hasJson = await fileManager.pathExists(jsonPath);
+    const hasJsonc = await fileManager.pathExists(jsoncPath);
+
+    // Determine key prefix preferences (with sensible defaults)
+    // Defaults: non-prefixed (agents = "dev", commands = "create-doc")
+    let useAgentPrefix = false;
+    let useCommandPrefix = false;
+
+    // Allow pre-configuration (if passed) to skip prompts
+    const pre = preConfiguredSettings && preConfiguredSettings.opencode;
+    if (pre && typeof pre.useAgentPrefix === 'boolean') useAgentPrefix = pre.useAgentPrefix;
+    if (pre && typeof pre.useCommandPrefix === 'boolean') useCommandPrefix = pre.useCommandPrefix;
+
+    // If no pre-config and in interactive mode, prompt the user
+    if (!pre) {
+      // Pause spinner during prompts if active
+      let spinnerWasActive = false;
+      if (spinner && spinner.isSpinning) {
+        spinner.stop();
+        spinnerWasActive = true;
+      }
+
+      try {
+        const resp = await inquirer.prompt([
+          {
+            type: 'confirm',
+            name: 'useAgentPrefix',
+            message:
+              "Prefix agent keys with 'bmad-'? (Recommended to avoid collisions, e.g., 'bmad-dev')",
+            default: false,
+          },
+          {
+            type: 'confirm',
+            name: 'useCommandPrefix',
+            message:
+              "Prefix command keys with 'bmad:tasks:'? (Recommended, e.g., 'bmad:tasks:create-doc')",
```

Comment:
Are different task being used by different agents? If that is not the case, maybe a better prefix would be `bmad:<agent_id>:<task>`.